squad/core/tasks/__init__.py

```diff
@@ -151,16 +153,17 @@ def __call__(self, version, environment_slug, metadata_file=None, metrics_file=N
 
         testrun = build.test_runs.create(
             environment=environment,
-            tests_file=tests_file,
-            metrics_file=metrics_file,
-            log_file=log_file,
             metadata_file=metadata_file,
             completed=completed,
             **metadata_fields
         )
+        testrun.tests_file_storage.save("testrun/%s/tests_file" % testrun.pk, File(StringIO(tests_file)))
+        testrun.metrics_file_storage.save("testrun/%s/metrics_file" % testrun.pk, File(StringIO(metrics_file)))
+        testrun.log_file_storage.save("testrun/%s/log_file" % testrun.pk, File(StringIO(log_file)))
 
         for f, data in attachments.items():
-            testrun.attachments.create(filename=f, data=data, length=len(data))
+            attachment = testrun.attachments.create(filename=f, length=len(data))
+            attachment.storage.save(f, File(BytesIO(data)))
```

Comment:
This is creating files in the root of storage. I fixed it locally, will send PR soon